lib/rules/no-trailing-spaces.js

```diff
@@ -10,7 +10,12 @@
 
 module.exports = function(context) {
 
-    var TRAILER = "[ \t\u00a0\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u200b\u2028\u2029\u3000]+$";
+    var BLANK_PREFIX = "[^( |\t)]",
+        TRAILER = "[ \t\u00a0\u2000\u2001\u2002\u2003\u2004\u2005\u2006\u2007\u2008\u2009\u200a\u200b\u2028\u2029\u3000]+$";
+
+    var options = context.options[0] || {},
+        skipBlankLines = +!!options.skipBlankLines; // Defaults to false
```

Comment:
Are you referring to TRAILER?  That was the original code.  I can remove if needed, though.